edgelet/aziot-edged/src/watchdog.rs

```diff
@@ -134,6 +140,34 @@ async fn create_and_start_agent(
     let mut env = agent_env(gen_id, settings, device_info);
     agent_spec.env_mut().append(&mut env);
 
+    let pinfo = if let Some(path) = settings.product_info() {
+        Ok(
+            crate::product_info::ProductInfo::try_load(path).map_err(|err| {
+                EdgedError::from_err(
+                    format!("malformed product info at path {}", path.display()),
+                    err,
+                )
+            })?,
```

Comment:
This can probably be checked earlier (maybe in Settings::new) and fail during startup, rather than reaching this point and failing.